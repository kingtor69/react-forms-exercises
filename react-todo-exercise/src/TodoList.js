import React, {useState} from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

const TodoList = () => {
  const [todoList, setTodoList] = useState([]);
  console.log(Array.isArray(todoList));

  const addTodo = (newTodo) => {
    setTodoList([
      ...todoList,
      newTodo
    ]);
  };

  const deleteTodo = (todoId) => {
    setTodoList(todoList.filter ((t, i) => (i !== todoId)));
  };

  return (
    <div className="TodoList">
      <NewTodoForm 
        addTodo={addTodo} 
      />
      <ul>
        {todoList.map((todo, i) => (
          <Todo 
            todo={todo}
            key={i}
            id={i}
            deleteTodo={deleteTodo} 
          />
        ))}
      </ul>
    </div>
  );
};

export default TodoList;
