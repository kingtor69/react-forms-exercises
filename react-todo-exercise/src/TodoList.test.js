import { render } from '@testing-library/react';
import TodoList from './TodoList';

it('renders without crashing', () => {
  render(<TodoList />);
});

it('renders snapshot', () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});
