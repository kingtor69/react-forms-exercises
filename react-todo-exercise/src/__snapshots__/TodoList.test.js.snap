// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <form
      class="NewTodoForm"
    >
      <label
        for="new-todo"
      >
        New todo:
      </label>
      <input
        id="new-todo"
        placeholder="new todo"
        type="text"
        value=""
      />
      <button>
        Create new todo
      </button>
    </form>
    <ul />
  </div>
</DocumentFragment>
`;
